src/main.c

```diff
@@ -1024,6 +1024,12 @@ common_init_2(mparm_T *paramp)
 #ifdef FEAT_SIGNS
     init_signs();
 #endif
+
+#ifdef FEAT_QUICKFIX
+    // initialize global quickfix list
+    if (qf_init_global_stack() == FAIL)
+	emsg(_(e_no_quickfix_stack));
```

Comment:
Similar to my previous comment, I think it would make sense to not show an error message  when Vim starts, but to show one when we try to use quickfix.